### ISSUE TITLE
regression in conda.fetch.download: it now ungzips the zlib tar.gz

### ISSUE THREAD

**Issue by mwiebe** [conda/conda#3540]
This causes an MD5 error when using conda-build on a recipe applying the MD5 documented from the zlib website.

```
>wget http://zlib.net/zlib-1.2.8.tar.gz
SYSTEM_WGETRC = c:/progra~1/wget/etc/wgetrc
syswgetrc = C:\Gow/etc/wgetrc
--2016-09-27 13:26:53--  http://zlib.net/zlib-1.2.8.tar.gz
Resolving zlib.net... 69.73.132.10
Connecting to zlib.net|69.73.132.10|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 571091 (558K) [application/x-gzip]
Saving to: `zlib-1.2.8.tar.gz'

100%[===================================================================================================================================>] 571,091      700K/s   in 0.8s

2016-09-27 13:26:55 (700 KB/s) - `zlib-1.2.8.tar.gz' saved [571091/571091]

>python
Python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from conda import fetch
>>> help(fetch.download)
Help on function download in module conda.fetch:

download(url, dst_path, session=None, md5=None, urlstxt=False, retries=None)

>>> fetch.download('http://zlib.net/zlib-1.2.8.tar.gz', 'zlib2.tar.gz')
>>> exit()

>ls -al *.tar.gz
-rw-rw-rw-  1 mwiebe 0  571091 2013-04-28 17:33 zlib-1.2.8.tar.gz
-rw-rw-rw-  1 mwiebe 0 2795520 2016-09-27 13:27 zlib2.tar.gz
```

**Comment by mwiebe** [comment]
The zlib website includes a note about this:

> Note that if your MD5 checksum does not match, you should check the length of the downloaded .tar.gz file. If it's something like 2.7 MB, then your browser decompressed it without telling you (how nice!), but nevertheless left the .gz in the name. In that case, please don't email me to tell me that my website must have been compromised since the MD5 is wrong, because it isn't and it hasn't.

**Comment by kalefranz** [comment]
We've done a lot of work to clean this up.  Should be fixed in 4.3, if not 4.2.  Possibly related discussion in https://github.com/conda/conda/pull/3991.
